subsys/secure_storage/src/its/transform/aead_get.c

```diff
@@ -0,0 +1,141 @@
+/* Copyright (c) 2024 Nordic Semiconductor
+ * SPDX-License-Identifier: Apache-2.0
+ */
+#include <zephyr/secure_storage/its/transform/aead_get.h>
+#include <zephyr/drivers/hwinfo.h>
+#include <zephyr/init.h>
+#include <zephyr/logging/log.h>
+#include <psa/crypto.h>
+#include <string.h>
+#include <sys/types.h>
+
+LOG_MODULE_DECLARE(secure_storage, CONFIG_SECURE_STORAGE_LOG_LEVEL);
+
+#ifdef CONFIG_SECURE_STORAGE_ITS_TRANSFORM_AEAD_SCHEME_AES_GCM
+#define PSA_KEY_TYPE PSA_KEY_TYPE_AES
+#define PSA_ALG PSA_ALG_GCM
+#elif defined(CONFIG_SECURE_STORAGE_ITS_TRANSFORM_AEAD_SCHEME_CHACHA20_POLY1305)
+#define PSA_KEY_TYPE PSA_KEY_TYPE_CHACHA20
+#define PSA_ALG PSA_ALG_CHACHA20_POLY1305
+#endif
+#ifndef CONFIG_SECURE_STORAGE_ITS_TRANSFORM_AEAD_SCHEME_CUSTOM
+void secure_storage_its_transform_aead_get_scheme(psa_key_type_t *key_type, psa_algorithm_t *alg)
+{
+	*key_type = PSA_KEY_TYPE;
+	*alg = PSA_ALG;
+}
+#endif /* !CONFIG_SECURE_STORAGE_ITS_TRANSFORM_AEAD_SCHEME_CUSTOM */
+
+#ifndef CONFIG_SECURE_STORAGE_ITS_TRANSFORM_AEAD_KEY_PROVIDER_CUSTOM
+
+#define SHA256_OUTPUT_SIZE 32
+BUILD_ASSERT(SHA256_OUTPUT_SIZE == PSA_HASH_LENGTH(PSA_ALG_SHA_256));
+BUILD_ASSERT(SHA256_OUTPUT_SIZE >= CONFIG_SECURE_STORAGE_ITS_TRANSFORM_AEAD_KEY_SIZE);
+
+static psa_status_t hash_data_into_key(size_t data_len, const void *data, uint8_t key[
+				       static CONFIG_SECURE_STORAGE_ITS_TRANSFORM_AEAD_KEY_SIZE])
+{
+	size_t hash_len;
+
+#if CONFIG_SECURE_STORAGE_ITS_TRANSFORM_AEAD_KEY_SIZE == SHA256_OUTPUT_SIZE
```

Comment:
Just a note: this could done by `if` as compiler is able to optimize dead code when sees comparison of literals.